reactor-netty-core/src/test/java/reactor/netty/transport/ProxyProviderTest.java

```diff
@@ -204,6 +207,171 @@ void shouldNotCreateProxyProviderWithMissingRemoteHostInfo() {
 				.withMessage("Neither address nor host is specified");
 	}
 
+	@Test
+	void proxyFromSystemProperties_nullProxyProviderIfNoHostnamePropertySet() {
+		Properties properties = new Properties();
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider).isNull();
+	}
+
+	@Test
+	void proxyFromSystemProperties_proxyProviderIsNotNullWhenHttpHostSet() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.HTTP_PROXY_HOST, "host");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+
+		assertThat(provider.getType()).isEqualTo(ProxyProvider.Proxy.HTTP);
+		assertThat(provider.getAddress().get().getHostName()).isEqualTo("host");
```

Comment:
Do we need to resolve the host? If no - then let's use `getHostString()`.

Reply:
Good catch, replaced with `getHostString()`. Thank you.